src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/CosmosDb/CosmosCommon.cs

```diff
@@ -0,0 +1,174 @@
+        public const string Major3MinorX = "3";
+
+        public const string IntegrationName = nameof(IntegrationIds.CosmosDb);
+
+        private const string OperationName = "cosmosdb.query";
+        private const string ServiceName = "cosmosdb";
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(CosmosCommon));
+
+        public static CallTargetState CreateContainerCallStateExt<TTarget, TQueryDefinition>(TTarget instance, TQueryDefinition queryDefinition)
+        {
+            return CreateCosmosDbCallState(GetContainerPropterties, instance, queryDefinition);
+        }
+
+        public static CallTargetState CreateDatabaseCallStateExt<TTarget, TQueryDefinition>(TTarget instance, TQueryDefinition queryDefinition)
+        {
+            return CreateCosmosDbCallState(GetDatabasePropterties, instance, queryDefinition);
+        }
+
+        public static CallTargetState CreateClientCallStateExt<TTarget, TQueryDefinition>(TTarget instance, TQueryDefinition queryDefinition)
+        {
+            return CreateCosmosDbCallState(GetClientPropterties, instance, queryDefinition);
+        }
+
+        private static CallTargetState CreateCosmosDbCallState<TTarget, TQueryDefinition>(Func<object, Tuple<string, string, string>> extractProperties, TTarget instance, TQueryDefinition queryDefinition)
+        {
+            if (!Tracer.Instance.Settings.IsIntegrationEnabled(IntegrationName))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return CallTargetState.GetDefault();
+            }
+
+            try
+            {
+                string query;
+                if (queryDefinition is string queryDefinitionString)
+                {
+                    query = queryDefinitionString;
+                }
+                else
+                {
+                    var success = queryDefinition.TryDuckCast<QueryDefinitionStruct>(out var queryDefinitionObj);
+                    query = queryDefinitionObj.QueryText;
+                }
+
+                var tracer = Tracer.Instance;
+
+                var parent = tracer.ActiveScope?.Span;
+
+                if (parent != null &&
+                    parent.Type == SpanTypes.Sql &&
+                    parent.GetTag(Tags.DbType) == "cosmosdb" &&
+                    parent.ResourceName == query)
+                {
+                    // we are already instrumenting this,
+                    // don't instrument nested methods that belong to the same stacktrace
+                    return new CallTargetState(null);
+                }
+
+                var properties = extractProperties(instance);
+
+                string containerId = properties.Item1;
+                string databaseId = properties.Item2;
+                string endpoint = properties.Item3;
+
+                if (endpoint == null)
+                {
+                    Console.WriteLine();
```

Comment:
Is this left over from previous testing code?

Reply:
Yes, removed.